src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageService.Factory.cs

```diff
@@ -22,13 +22,7 @@ internal partial class Factory(
         public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
         {
             var textFactory = workspaceServices.GetRequiredService<ITextFactoryService>();
-
-            // MemoryMapped files which are used by the TemporaryStorageService are present in .NET Framework (including Mono)
-            // and .NET Core Windows. For non-Windows .NET Core scenarios, we can return the TrivialTemporaryStorageService
-            // until https://github.com/dotnet/runtime/issues/30878 is fixed.
-            return PlatformInformation.IsWindows || PlatformInformation.IsRunningOnMono
-                ? new TemporaryStorageService(workspaceThreadingService, textFactory)
-                : TrivialTemporaryStorageService.Instance;
+            return new TemporaryStorageService(workspaceThreadingService, textFactory);
```

Comment:
i removed Trivial entirely..  Now there's just the real service.  And the real service *can* always dump data to a mmf.  However, it can't necessarily *share* that data cross processes.  That capability is Windows/Mono only.  but the only scenario where we share across processes it the VS hybrid host/OOP model, which is win only.  so thsi should be fine.